### PR TITLE
fix: DateFNS locale fallback mapping

### LINKED ISSUE #147: generateConfig for date-fns throws error for some locales
The `generateConfig` methods use [`dealLocal()`](https://github.com/react-component/picker/blob/master/src/generate/dateFns.ts#L27) function to get `date-fns` specific locale data for the provided language, but the implementation fails in some cases because `rc-picker` locales names **do not exactly match** with `date-fns` ones.

For example Italian:
- `rc-picker` use [`it_IT`](https://github.com/react-component/picker/blob/master/src/locale/it_IT.ts#L4) as locale name;
- `dealLocal()` converts it into `itIT`;
- `generateConfig.getWeekFirstDay()` method [tries to load](https://github.com/react-component/picker/blob/master/src/generate/dateFns.ts#L68) `date-fns` locale data with an `itIT` key, but [**it does not exists**](https://github.com/date-fns/date-fns/blob/master/src/locale/index.js#L42) and this leads to `clone` constant to be `undefined`.

The only solution is to replace the current `dealLocal()` implementation with a static map of locales names and prevent errors with a default value:
```ts
const DEFAULT_LOCALE = 'enUS';

const LOCALES_MAP = {
  'en_US': 'enUS',
  'it_IT': 'it',
  'es_ES': 'es',

  // and so on...
}

const dealLocal = (str: string): string => LOCALES_MAP[str] || DEFAULT_LOCALE;
```

### PR DESCRIPTION
**Pull request by zombieJ**
ref https://github.com/react-component/picker/pull/289#issuecomment-1063640635

close #147
close #289 
close ant-design/ant-design#47339
close ant-design/ant-design#47600

### CODE REVIEW

File: src/generate/dateFns.ts [hunk 1]
```diff
@@ -23,16 +23,15 @@ import {
...
+import * as locales from 'date-fns/locale';
 import type { GenerateConfig } from '.';
 
-const localeMap: Record<string, string> = {
-  ko_KR: 'ko'
-};
-
-const dealLocal = (str: string) => {
-  return localeMap[str] || str.replace(/_/g, '');
+const getLocale = (locale: string): Locale => {
```
Comment (yoyo837) [inline src/generate/dateFns.ts:31]: 👍

File: tests/generate.spec.tsx [hunk 2]
```diff
@@ -309,3 +309,13 @@ describe('Generate:dayjs', () => {
...
   });
 });
+
+describe('Generate:date-fns', () => {
+  it('getWeekFirstDay', () => {
+    expect(dateFnsGenerateConfig.locale.getWeekFirstDay('en_US')).toEqual(0);
+    expect(dateFnsGenerateConfig.locale.getWeekFirstDay('zh_CN')).toEqual(1);
+    expect(dateFnsGenerateConfig.locale.getWeekFirstDay('ar_EG')).toEqual(0);
+    expect(dateFnsGenerateConfig.locale.getWeekFirstDay('ar_MA')).toEqual(1);
+    expect(dateFnsGenerateConfig.locale.getWeekFirstDay('ar')).toEqual(6);
```
Comment (yoyo837) [inline tests/generate.spec.tsx:319]: 再多加几个 issue 或者 pr 里提到的语言？ ko_KR、it_IT、fr_FR。

**Comment by yoyo837** [comment]
@nnnnoel Please help to check if this breaks #603.

**Comment by nnnnoel** [comment]
@yoyo837 It doesn't breaks my previous PR 🙇

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**